okteto/okteto
# chore(deploy): better err on dev deployment fail

# Proposed changes

Fixes #3419

When the user overrides the deploy image, but the image is not accessible (eg. does not exist or the user doesn't have enough permissions to push) the CLI fails with an error that does not give away much. To figure out the issue, one has to run again enabling debug logs.

Proposed change:
1. propagate the manifest from the deployOptions to the buildOptions so that we can know the image name which failed
2. always wrap the current error with its nested cause.


### How to test

1. Clone `movies-frontend`
```bash
git clone git@github.com:okteto/movies-frontend.git
```
2. Change the `deploy` block of the Okteto manifest to 
```yaml
deploy:
  image: ciaociaociaociaociaociao/ciao:1
  commands:
    - name: Deploy Frontend
      command: helm upgrade --install movies-frontend chart --set image=${OKTETO_BUILD_FRONTEND_IMAGE}
```
3. Build the CLI on this branch
4. Run `okteto deploy` 
5. Observe the error being: `Error during development environment deployment: error building image 'ciaociaociaociaociaociao/ciao:1': Access Denied (the repository does not exist or you are not authorized to push a new image).`

| Before | After |
|--------|-------|
|  <img width="350" height="auto" src="https://user-images.githubusercontent.com/2318450/226758344-0499ac42-8520-4350-9c44-dac45cc8a4ba.png" />      |       <img width="350" height="auto" src="https://user-images.githubusercontent.com/2318450/226758303-189f0040-ef49-45d7-9d4c-5d8b2a4311bb.png" /> |
